src/System.Text.Json/tests/Serialization/JsonElementTests.cs

```diff
@@ -0,0 +1,143 @@
+                String = JsonDocument.Parse(@"""Hello""").RootElement.Clone();
+                Array = JsonDocument.Parse(@"[2, false, true, ""Goodbye""]").RootElement.Clone();
+                Object = JsonDocument.Parse(@"{}").RootElement.Clone();
+            }
+
+            public void Verify()
+            {
+                Assert.Equal(JsonValueType.Number, Number.Type);
+                Assert.Equal("1", Number.ToString());
+                Assert.Equal(JsonValueType.True, True.Type);
+                Assert.Equal("True", True.ToString());
+                Assert.Equal(JsonValueType.False, False.Type);
+                Assert.Equal("False", False.ToString());
+                Assert.Equal(JsonValueType.String, String.Type);
+                Assert.Equal("Hello", String.ToString());
+                Assert.Equal(JsonValueType.Array, Array.Type);
+                JsonElement[] elements = Array.EnumerateArray().ToArray();
+                Assert.Equal(JsonValueType.Number, elements[0].Type);
+                Assert.Equal("2", elements[0].ToString());
+                Assert.Equal(JsonValueType.False, elements[1].Type);
+                Assert.Equal("False", elements[1].ToString());
+                Assert.Equal(JsonValueType.True, elements[2].Type);
+                Assert.Equal("True", elements[2].ToString());
+                Assert.Equal(JsonValueType.String, elements[3].Type);
+                Assert.Equal("Goodbye", elements[3].ToString());
+                Assert.Equal(JsonValueType.Array, Array.Type);
+                Assert.Equal(JsonValueType.Object, Object.Type);
+                Assert.Equal("{}", Object.ToString());
+            }
+        }
+
+        [Fact]
+        public void SerializeJsonElementArray()
+        {
+            var obj = JsonSerializer.Parse<JsonElementArrayClass>(JsonElementArrayClass.s_json);
+            obj.Verify();
+            string reserialized = JsonSerializer.ToString(obj);
+
+            // Properties in the exported json will be in the order that they were reflected, doing a quick check to see that
+            // we end up with the same length (i.e. same amount of data) to start.
+            Assert.Equal(JsonElementArrayClass.s_json.StripWhitespace().Length, reserialized.Length);
+
+            // Shoving it back through the parser should validate round tripping.
+            obj = JsonSerializer.Parse<JsonElementArrayClass>(reserialized);
+            obj.Verify();
+        }
+
+        public class JsonElementArrayClass : ITestClass
+        {
+            public JsonElement[] Array { get; set; }
+
+            public static readonly string s_json =
+                @"{" +
+                    @"""Array"" : [" +
+                        @"1, " +
+                        @"true, " +
+                        @"false, " +
+                        @"""Hello""" +
+                        // @"[2, false, true, ""Goodbye""], " +
```

Comment:
Is this intentionally commented out?

Reply:
Yes, because it doesn't work yet. I'll add the comment.